teamcity: add new CI job for PUA

shailendra-patel opened the pull request:
During each release, there is a need to run PUA.
This new CI job allows triggering the PUA workflow on an as-needed basis for a specific release. The job generates a JSON file as output, which is consumed by the benchmarking portal to display the results.

Epic: none

Release note: None

shailendra-patel committed:
teamcity: add new CI job for PUA

During each release, there is a need to run PUA.
This new CI job allows triggering the PUA workflow on an as-needed
basis for a specific release. The job generates a JSON file as output,
which is consumed by the benchmarking portal to display the results.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/151514)
<!-- Reviewable:end -->

shailendra-patel committed:
drtprod: update PUA yaml configs

This commits updates the PUA yaml configs to run via
CI in cockroach-ephemeral project.

Epic: none

Release note: None

srosenberg committed:
fix pua_run_impl.sh

shailendra-patel committed:
Set environment variable COCKROACH_ROACHPROD_INSECURE to false.

Epic: none

Release note: None

srosenberg reviewed (approved):
Nice work; left a couple of minor comments, otherwise LGTM.

shailendra-patel commented on pkg/cmd/drtprod/scripts/tpcc_run_multiregion.sh:
```diff
@@ -49,11 +50,12 @@ do
 export ROACHPROD_GCE_DEFAULT_PROJECT=$ROACHPROD_GCE_DEFAULT_PROJECT
-./roachprod sync
-PGURLS=\$(./roachprod load-balancer pgurl $CLUSTER | sed s/\'//g)
+export COCKROACH_ROACHPROD_INSECURE="${COCKROACH_ROACHPROD_INSECURE:-false}"
```
I am keeping it in favour of if someone want to use this script on insecure cluster, they can set this env variable to `true`.
